fairseq2n/src/fairseq2n/data/data_pipeline.h

```diff
@@ -108,6 +111,8 @@ using map_fn = std::function<data(data &&)>;
 
 using predicate_fn = std::function<bool(const data &)>;
 
+using index_generator_fn = std::function<std::size_t()>;
```

Comment:
Ideally this alias should reside in `multi_data_source.h`. It is not used by any of the public data pipeline ops and is mostly an implementation detail used by `sample` and `round_robin`.